changelog.d/8011.bugfix

```diff
@@ -0,0 +1 @@
+Fix a long-standing bug which caused two copies of some log lines to be written when synctl was used along with a MemoryHandler logger.
```

Comment:
tbh the refactor was mostly to support this fix. have decided to merge the newsfiles.